src/ui/GridUI.java

```diff
@@ -84,6 +84,8 @@ public void renderZombies() {
 		
 		if (zombiePanel != null) {
 			this.remove(zombiePanel);
+			repaint();
+			revalidate();
```

Comment:
I tried your changes and it seems like you don't need `revalidate()`.

Reply:
Done